File: Nii_Raw_Interconversion/savenii.m
%load directory in question (change this to match your file structure)
imgdir = dir('./*/*.raw');
xx = 512; yy = 512; zz = 176;

%go through directory
for i = 1 : length(imgdir)
    
    %this matches ID in my file names - might need to change
    file_name = imgdir(i).name(1:end-4);
    
    %open raw file
    fid = fopen(fullfile(imgdir(i).folder, imgdir(i).name));
    rawdata = fread(fid, inf, '*double');
    rawdata = reshape(rawdata, xx, yy, zz);

    nii = make_nii(rawdata);
    save_nii(nii, fullfile(imgdir(i).folder, strcat(file_name, '.nii')));

    %load back from nii
    imgs = load_untouch_nii(fullfile(imgdir(i).folder, imgdir(i).name)).img;

    %Test that it worked by opening last raw file:
    figure; imshow(imgs(:,:,floor(zz/2)), []);
    
end

File: Nii_Raw_Interconversion/savenii.py
import glob
import nibabel as nib
import numpy as np

imgdir = glob.glob('./*/*.raw')
xx, yy, zz = 512, 512, 176

# go through directory
for i in range(len(imgdir)):
    # this matches ID in my file names - might need to change
    file_name = imgdir[i][:-4]

    # open raw file
    with open(imgdir[i], 'rb') as f:
        rawdata = np.fromfile(f, dtype=np.double)
        rawdata = np.reshape(rawdata, (xx, yy, zz))

    nii = nib.Nifti1Image(rawdata, np.eye(4))
    nib.save(nii, imgdir[i][:-4] + '.nii')

    # load back from nii
    imgs = nib.load(imgdir[i][:-4] + '.nii').get_fdata()

    # Test that it worked by opening last raw file:
    plt.imshow(imgs[:, :, int(zz/2)], cmap='gray')
    plt.show()
